v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerSimpleIT.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (C) 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates;
+
+import static org.apache.beam.it.truthmatchers.PipelineAsserts.assertThatResult;
+
+import com.google.cloud.teleport.metadata.SkipDirectRunnerTest;
+import com.google.cloud.teleport.metadata.TemplateIntegrationTest;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.beam.it.common.PipelineLauncher;
+import org.apache.beam.it.common.PipelineOperator;
+import org.apache.beam.it.conditions.ChainedConditionCheck;
+import org.apache.beam.it.gcp.spanner.conditions.SpannerRowsCheck;
+import org.apache.beam.it.gcp.spanner.matchers.SpannerAsserts;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** A Simple Integration test for {@link DataStreamToSpanner} Flex template. */
+@Category({TemplateIntegrationTest.class, SkipDirectRunnerTest.class})
+@TemplateIntegrationTest(DataStreamToSpanner.class)
+@RunWith(JUnit4.class)
+public class DataStreamToSpannerSimpleIT extends DataStreamToSpannerITBase {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DataStreamToSpannerSimpleIT.class);
+
+  private static final String TABLE1 = "Users";
+  private static final String TABLE2 = "Movie";
+
+  private static final String SESSION_FILE_RESOURCE =
+      "DataStreamToSpannerSimpleIT/mysql-session.json";
+
+  private static ArrayList<DataStreamToSpannerSimpleIT> testInstances = new ArrayList<>();
+  private static PipelineLauncher.LaunchInfo jobInfo;
+
+  /**
+   * Setup resource managers and Launch dataflow job once during the execution of this test class.
+   *
+   * @throws IOException
+   */
+  @Before
+  public void setUp() throws IOException {
+    // Prevent cleaning up of dataflow job after a test method is executed.
+    skipBaseCleanup = true;
+    testInstances.add(this);
```

Comment:
Isn't @Before executes once for every @Test? If so, wouldn't testInstances have duplicate entires? Since we marked cleaUp as @AfterClass should this method be marked @BeforeClass?

Reply:
I tried to but could not change @before to @beforeclass. @beforeclass should be a static method, and the setup relies on many instance variables defined in TemplatesTestBase class (provided by the framework). So, making this a static method was not possible.

I have changed the variable type to Set instead of List to avoid duplication.

Since the set of instances is saved, we could do cleanup in the static @afterclass method.